### DISCUSSION TITLE
Dropdown prevent closing after click

### DISCUSSION THREAD

**Discussion by bastigerner** [Megabit/Blazorise#2278] [Q&A]
Hey,

is it possible to prevent a dropdown from closing after I have clicked something inside it?

![image](https://user-images.githubusercontent.com/10349466/116684109-d2265d80-a9b0-11eb-8727-ae41cd1bc1ec.png)

I am trying to create a multiselect filter and want it to stay open until I click the DropdownToggle again.

Kind regards

**Comment by stsrki** [comment]
It is not possible currently with Blazorise.

You could make it possible if you create a custom toggle button that you can use to toggle the menu.

**CustomDropdownToggle.razor**

```cshtml
<button type="button" class="@($"btn dropdown-toggle btn-primary {(ParentDropdown.Visible ? "show":"")}")" data-toggle="dropdown" aria-expanded="ParentDropdown.Visible"  @onclick="@ClickHandler">
    @ChildContent
</button>

@code{
  [CascadingParameter] protected Dropdown ParentDropdown { get; set; }

  Task ClickHandler(){
    ParentDropdown?.Toggle();
    
    return Task.CompletedTask;
  }
}
}
```

**Reply by stsrki** [reply]
PS. modify if needed :)

**Reply by bastigerner** [reply]
Thanks going to test it soon!

**Reply by bastigerner** [reply]
Simply just remove 

`data-toggle="dropdown"`

did the job for me. Just add a little bit of margin to the `dropdown-menu` and ur good to go I guess.
